src/plugins/data_source/server/plugin.ts

```diff
@@ -104,6 +106,11 @@ export class DataSourcePlugin implements Plugin<DataSourcePluginSetup, DataSourc
       return dataSourcePluginStart.getAuthenticationMethodRegistery();
     });
 
+    const customApiSchemaRegistryPromise = core.getStartServices().then(([, , selfStart]) => {
```

Comment:
do we have a chance to test this with any plugin and see if there any issue with plugin integration. Thanks

Reply:
Yes, tests have been done using query bench and it is in the recorded video